.github/workflows/build-binary.yml

```diff
@@ -0,0 +1,27 @@
+name: Build Binary
+
+on: { push: { branches-ignore: [main, production] } }
+
+jobs:
+  preprod-agent-deployment:
+    name: Build and upload agent
+    runs-on: ubuntu-latest
+    container: golang:1.16
+    env:
+      BASE_NAME: spacelift-vcs-agent
+      BIN_DIR: build
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@master
+
+      - name: parse short SHA
+        id: vars
+        run: |
+          echo ::set-output name=sha::$(git rev-parse --short=8 ${{ github.sha }})
+
+      - name: Build Spacelift VCS Agent
+        run: go build -a -tags netgo -ldflags '-s -w -extldflags "-static"' -trimpath -o $BIN_DIR/$BASE_NAME ./cmd/spacelift-vcs-agent
+        env:
+          CGO_ENABLED: 0
+          SHORT_SHA: ${{ steps.vars.outputs.sha }}
```

Comment:
It looks like this isn't used. Were you meaning to pass it in as a linker flag?

Reply:
You're right! I was just copy pasting what was previously here.

I'll first bring it all to work, and then add a version variable we can inject through the linker.